helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableReplica.java

```diff
@@ -107,6 +107,18 @@ public int compareTo(AssignableReplica replica) {
     return 0;
   }
 
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
```

Comment:
if (obj == null || !(obj instanceof AssignableReplica)) {
  return false;
}
return compareTo((AssignableReplica) obj) == 0;

Reply:
This is mainly for cleaner code. And better to ensure the same code style as the other equals() in the Helix project.